nexus/reconfigurator/planning/src/blueprint_builder/builder.rs

```diff
@@ -1982,18 +1982,28 @@ impl<'a> BlueprintBuilder<'a> {
     }
 
     fn zone_image_artifact(
-        repo: Option<&TufRepoDescription>,
+        repo: &TargetReleaseDescription,
         zone_kind: ZoneKind,
     ) -> BlueprintZoneImageSource {
-        repo.and_then(|repo| {
-            repo.artifacts
-                .iter()
-                .find(|artifact| {
-                    zone_kind.is_control_plane_zone_artifact(&artifact.id)
-                })
-                .map(BlueprintZoneImageSource::from_available_artifact)
-        })
-        .unwrap_or(BlueprintZoneImageSource::InstallDataset)
+        let repo = match repo {
+            TargetReleaseDescription::Initial => {
+                // If `repo` describes the initial target release, we have no
+                // TUF repo at all; all zones are sourced from their install
+                // dataset.
+                return BlueprintZoneImageSource::InstallDataset;
+            }
+            TargetReleaseDescription::TufRepo(repo) => repo,
+        };
+
+        repo.artifacts
+            .iter()
+            .find(|artifact| {
+                zone_kind.is_control_plane_zone_artifact(&artifact.id)
+            })
+            .map(BlueprintZoneImageSource::from_available_artifact)
+            // TODO-john this should be an error - this is a repo that's missing
+            // an artifact of a known zone kind
+            .unwrap_or(BlueprintZoneImageSource::InstallDataset)
```

Comment:
Should we do this before merging? Or panic, or something?

This seems quite bad, especially if we perform an upgrade that unintentionally causes a downgrade